### ISSUE TITLE
Feature request: Access to results

### ISSUE THREAD

**Issue by markdall** [tqdm/tqdm#860]
- [X ] I have marked all applicable categories:
    + [X] new feature request

tqdm gives nice human readable information about iterations per second. It would be great if we could access that information in order to log it, for times when we want to check the performance of a thing over time. 

From this example: https://stackoverflow.com/questions/56677267/tqdm-extract-time-passed-time-remaining 

 0%|          | 0/200 [00:00<?, ?it/s]
  4%|▎         | 7/200 [00:00<00:02, 68.64it/s]
  8%|▊         | 16/200 [00:00<00:02, 72.87it/s]
 12%|█▎        | 25/200 [00:00<00:02, 77.15it/s]
 17%|█▋        | 34/200 [00:00<00:02, 79.79it/s]
 22%|██▏       | 43/200 [00:00<00:01, 79.91it/s]
 26%|██▌       | 52/200 [00:00<00:01, 80.23it/s]
 30%|███       | 61/200 [00:00<00:01, 82.13it/s]
....
100%|██████████| 200/200 [00:02<00:00, 81.22it/s]

I'd like to be able to retrieve 81.22, which is the last value for iterations per second. I have another script that uses tqdm and shows me k/s for a download, and I'd love to be able to log that over time. The goal is very much to have the values match what is shown (by tqdm) exactly, vs. calculating it separately and maybe coming close.

**Comment by casperdcl** [comment]
Use the `tqdm` object's `.format_dict` property

**Comment by casperdcl** [comment]
see #562 (#482 #494 #660 #172 #656 #587)

**Comment by markdall** [comment]
Thanks so much! This was tough to know the right search terms for.

**Comment by casperdcl** [comment]
yes but I agree that there's quite a bit of info that `format_meter` calculates internally from `format_dict` which may be nice to expose in some way. As of now it still seems that calling `str()` around a `tqdm` object and parsing the output manually might be the best way to go which seems weird.

**Comment by markdall** [comment]
I'm just not great at it and for some reason keep ending up with a number that's different than the last displayed number. But it's much better than nothing, and I really appreciate the help. Thanks!

**Comment by graue70** [comment]
> yes but I agree that there's quite a bit of info that `format_meter` calculates internally from `format_dict` which may be nice to expose in some way.

This would be very useful.

I have something like the following at the moment:
```python
logger.info(f"Reading things from '{file}'")
with file.open("rb") as f:
    progress = tqdm(
        f, desc="Reading things", unit="lines", unit_scale=True
    )
    for line_number, line in enumerate(progress):
        # do something with line

logger.info(
    f"Read {progress.format_dict['n']} things in"
    f" {progress.format_interval(progress.format_dict['elapsed'])}"
    f" ({(progress.format_dict['n'] / progress.format_dict['elapsed']):.2f}lines/s)"
)
```

Besides this being unreadable code, the unit is not scaled in the logging statement.

**Comment by casperdcl** [comment]
> the unit is not scaled in the logging statement.

You can use `progress.format_sizeof()` for numbers the same way you use `progress.format_interval()` for seconds.

**Comment by graue70** [comment]
> You can use `progress.format_sizeof()` for numbers the same way you use `progress.format_interval()` for seconds.

Wow thanks. That's awesome. Who would have thought there is some non-documented functionality hidden away in the package? :)

In that case it's only about unreadable code or the ease of extracting the relevant information.

**Comment by casperdcl** [comment]
> Wow thanks. That's awesome. Who would have thought there is some non-documented functionality hidden away in the package? :)

well there's https://tqdm.github.io/docs/tqdm/#format_sizeof

**Comment by graue70** [comment]
> well there's https://tqdm.github.io/docs/tqdm/#format_sizeof

You're absolutely right, but I only ever looked at https://github.com/tqdm/tqdm#documentation where it doesn't appear. I find it confusing to have multiple sources of documentation with varying content or levels of detail.
